sdk/data/azcosmos/emulator_cosmos_item_test.go

```diff
@@ -503,6 +503,213 @@ func TestItemIdEncodingComputeGW(t *testing.T) {
 	verifyEncodingScenario(t, container, "ComputeGW-IdWithUnicodeCharacters", "WithUnicode鱀", http.StatusCreated, http.StatusOK, http.StatusOK, http.StatusNoContent)
 }
 
+func TestItemCRUDHierarchicalPartitionKey(t *testing.T) {
+	emulatorTests := newEmulatorTests(t)
+	client := emulatorTests.getClient(t, newSpanValidator(t, spanMatcher{
+		ExpectedSpans: []string{},
+	}))
+
+	database := emulatorTests.createDatabase(t, context.TODO(), client, "itemCRUDHierarchicalPartitionKey")
+	defer emulatorTests.deleteDatabase(t, context.TODO(), database)
+	properties := ContainerProperties{
+		ID: "aContainer",
+		PartitionKeyDefinition: PartitionKeyDefinition{
+			Paths:   []string{"/id", "/type"},
+			Kind:    PartitionKeyKindMultiHash,
+			Version: 2,
+		},
+	}
+
+	_, err := database.CreateContainer(context.TODO(), properties, nil)
+	if err != nil {
+		t.Fatalf("Failed to create container: %v", err)
+	}
+
+	container, err := database.NewContainer("aContainer")
+	if err != nil {
+		t.Fatalf("Failed to get container: %v", err)
+	}
+
+	itemAlpha := map[string]interface{}{
+		"id":    "1",
+		"type":  "alpha",
+		"value": "0",
+	}
+
+	itemBeta := map[string]interface{}{
+		"id":    "1",
+		"type":  "beta",
+		"value": "0",
+	}
```

Comment:
I don't think the documents can have the same ID, even if the partition key differs

Reply:
They should be able to.  Cosmos DB ensures pk + id is unique, but id doesn't have to be unique across partitions.

Reply:
Ah yeah, you're right. I was misreading some docs. I just verified and it does indeed work as long as the partition key is different. I expect the remaining test issue is the missing partition key below then.